### PR TITLE
[2574] added validation to detect missing StarReviews, marked it as required field, and error on not selected

### PR DESCRIPTION
**Pull request by pablobentivengo**
#2574 added validation to detect missing StarReviews, marked it as required field, and error on not selected

**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/2574
* Fixes https://github.com/scandipwa/scandipwa/issues/4130

**Problem:**
* Review form was not validation StarReviews as required

**In this PR:**
* For issue #2574 Added validation on submitting the form in ProductReviewFormContainer, renderStarErros function in ProductReviewFormComponent, and finally added the required "*".
* For issue #4130 added max-width to ReviewForm-Star Component

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
Issue 1: fixed
Issue 2: fixed
Issue 3: in RTL mode titles shown from left side, but should be from right side
Same logic for stars related to titles
image

**Comment by Elena-Evsa** [comment]
Issue 1: fixed
Issue 2: fixed
Issue 3: fixed
Issue 4: in LTR direction, quantity of stars increasing in wrong direction from right to left
https://vimeo.com/668142577/77efe210ee
Issue 5: RTL: message 'This field is required' situated from the right side of stars
image

File: packages/scandipwa/src/component/ReviewStar/ReviewStar.style.scss [hunk 1]
```diff
@@ -39,7 +41,7 @@
         border: 0;
     }
 
-    &:first-of-type {
+    /* &:first-of-type {
```
Comment (riha112) [inline packages/scandipwa/src/component/ReviewStar/ReviewStar.style.scss:44]: delete commented-out lines, no need to keep them.

File: packages/scandipwa/src/component/ReviewStar/ReviewStar.style.scss [hunk 2]
```diff
@@ -9,6 +9,8 @@
     border: 0;
     transition: color 150ms ease-in;
     color: $white;
+    // stylelint-disable declaration-no-important
+    margin-block-start: 0 !important;
```
Comment (riha112) [inline packages/scandipwa/src/component/ReviewStar/ReviewStar.style.scss:13]: Do not use `!important`, instead use higher priority class to override the rule.

File: packages/scandipwa/src/component/ProductReviewForm/ProductReviewForm.style.scss [hunk 3]
```diff
@@ -82,6 +79,20 @@
...
+        color: var(--required-text-color);
+    }
+    
+    &-ErrorMessage {
+        color: var(--primary-error-color);
+        font-size: 12px;
+        text-align: start;
+    }
+
+
```
Comment (riha112) [inline packages/scandipwa/src/component/ProductReviewForm/ProductReviewForm.style.scss:95]: Delete one new line

File: packages/scandipwa/src/component/ProductReviewForm/ProductReviewForm.style.scss [hunk 4]
```diff
@@ -39,16 +39,12 @@
...
-        [dir="rtl"] & {
-            display: flex;
-            flex-direction: row-reverse;
-        }
-
+        display: flex;
+        flex-flow: row-reverse nowrap;
         margin-block-start: 12px;
-        margin-block-end: 16px;
+        // margin-block-end: 16px;
```
Comment (riha112) [inline packages/scandipwa/src/component/ProductReviewForm/ProductReviewForm.style.scss:45]: delete commented-out lines, no need to keep them.

**Review by riha112 — CHANGES_REQUESTED** [review]
Implementation for star/rating validation can be done much simpler via `FieldGroup`.

Instead of creating separate validation logic you can add/wrap:
```js
<FieldGroup
  validationRule={ {
      isRequired: true,
      selector: '[type="radio"]'
  } }
>
    <fieldset block="ProductReviewForm" elem="Rating"....
    ...
</FieldGroup>
```
in `ProductReviewForm.component.js` -> `renderReviewRating`.

Form will then automatically validate it for you on submit.
(More info should be available here: https://www.notion.so/scandiweb/5-1-0-Config-Bundle-Form-Validation-update-v1-0-5366e403c9644a09ad634c5ffaada7cf#1844715db672410ea2a054889eb264b2)

File: packages/scandipwa/src/component/ReviewStar/ReviewStar.style.scss [hunk 5]
```diff
@@ -38,8 +42,6 @@
     &:focus {
         border: 0;
     }
-
-    &:first-of-type {
-        margin-block-start: 12px;
-    }
 }
+
+
```
Comment (riha112) [inline packages/scandipwa/src/component/ReviewStar/ReviewStar.style.scss:47]: Extra new line, should be only one

**Review by riha112 — CHANGES_REQUESTED** [review]

**Review by riha112 — APPROVED** [review]

**Pull request merged**